### DISCUSSION TITLE
How to contribue to the project without Github account?

### DISCUSSION THREAD

**Discussion by ahangarha** [mastodon/mastodon#27905] [General]
### Motivation

As perhaps you know, Github in the name of US trade law, has put restrictions for certain countries residents including my country, Iran. This incident motivated me more to leave this platform. More than this, I think it is not good for FLOSS community to rely on a proprietary platform for developing FLOSS projects and contributing to them. So, I request you guys to move to another platform which itself is FLOSS, such as Gitlab.

Till then, how can I contribute to Mastodon without using Github account?

**Comment by tribela** [comment]
There are gitlab for mastodon: https://source.joinmastodon.org/  But mastodon itself is not on there.

**Comment by ahangarha** [comment]
Should I wait? Any plan to move there or make it active?

Any other way to open issue and send PR to the project?

در تاریخ جمعه ۱۳ سپتامبر ۲۰۱۹،‏ ۲۱:۰۱ Jeong Arm <notifications@github.com>
نوشت:

**Comment by Gargron** [comment]
I'm afraid that I have been unhappy with GitLab both in terms of missing features (I use "notifications" on GitHub heavily to keep up with things) and the maintenance cost (difficult to upgrade, upgrades needed frequently, deployment methods not following best practices) as well as the fact that hosting an open-to-the-world GitLab is open to spam and malicious usage and closing it is impossible if we want to accept new contributors.

For this reason, I do not want to ever move the main repository to GitLab. If you want to contribute to Mastodon without using GitHub, you can export your git commits from your local repository as patch files and send them to hello@joinmastodon.org. I will create a pull request on your behalf and the commits will be authored by you.

**Comment by trwnh** [comment]
Can I suggest Gitea instead of Gitlab? It's much simpler to host and upgrade, has a UX very similar to Github, and supports features like notifications and OAuth (among many others).
